### PR TITLE
ICU-20593 Buildtool: Implementing new "additive" mode.

### PR DESCRIPTION
**Pull request by sffc**
##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-20593
- [x] Updated PR title and link in previous line to include Issue number
- [ ] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

**Comment by sffc** [comment]
I should add this to the userguide.  I can do that in the same PR if you like.

**Comment by sffc** [comment]
Here's the design doc:

https://docs.google.com/document/d/1C-sfojPrBUR7XPIg-GfazYYGg7dcN0euK5AjkBW-Fzk/edit

**Comment by markusicu** [comment]
> I should add this to the userguide. I can do that in the same PR if you like.

Yes, please.

File: icu4c/source/data/buildtool/filtration.py [hunk 1]
```diff
@@ -247,15 +247,21 @@ def _preprocess_file_filters(requests, config):
...
         if "featureFilters" in json_data and category in json_data["featureFilters"]:
-            filters[category] = Filter.create_from_json(
-                json_data["featureFilters"][category]
-            )
-        elif "localeFilter" in json_data and category[-5:] == "_tree":
-            filters[category] = Filter.create_from_json(
-                json_data["localeFilter"]
-            )
+            filter_json = json_data["featureFilters"][category]
+        if filter_json == "include" and "localeFilter" in json_data and category[-5:] == "_tree":
```
Comment (markusicu) [inline icu4c/source/data/buildtool/filtration.py]: Please use category.endswith("_tree")

**Pull request merged**